saleor/graphql/account/mutations/staff.py

```diff
@@ -281,6 +283,22 @@ def _save_m2m(cls, info, instance, cleaned_data):
     def post_save_action(cls, info, instance, cleaned_input):
         info.context.plugins.staff_created(instance)
 
+    @classmethod
+    def get_instance(cls, info, **data):
+        object_id = data.get("id")
+        email = data.get("input", {}).get("email")
+
+        if (
+            not object_id
+            and email
+            and (
+                user := models.User.objects.filter(email=email, is_staff=False).first()
+            )
+        ):
+            info.context.send_notification = False  # type: ignore
```

Comment:
I'm not a fan of passing such one-time flags in context; I would consider overriding the `perform_mutation` method in this case and calling all the model mutation's steps manually, with some additional logic in between that would check whether the notification should be sent.